src/EditorFeatures/CSharpTest/SplitStringLiteral/SplitStringLiteralCommandHandlerTests.cs

```diff
@@ -744,5 +752,56 @@ static void Main(string[] args)
 }",
             verifyUndo: false);
         }
+
+        [WorkItem(39040, "https://github.com/dotnet/roslyn/issues/39040")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitStringLiteral)]
+        public void TestMultiCaretSingleLine()
+        {
+            TestHandled(
+@"class C
+{
+    void M()
+    {
+        var v = ""now is [||]the ti[||]me"";
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        var v = ""now is "" +
+            ""[||]the ti"" +
+            ""[||]me"";
+    }
+}");
+        }
+
+        [WorkItem(39040, "https://github.com/dotnet/roslyn/issues/39040")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitStringLiteral)]
+        public void TestMultiCaretMultiLines()
+        {
+            TestHandled(
+@"class C
+{
+    string s = ""hello w[||]orld"";
+
+    void M()
+    {
+        var v = ""now is [||]the ti[||]me"";
+    }
+}",
+@"class C
+{
+    string s = ""hello w"" +
+        ""[||]orld"";
+
+    void M()
+    {
+        var v = ""now is "" +
+            ""[||]the ti"" +
+            ""[||]me"";
```

Comment:
can you have a test where we do this in different styles of strings. i.e.  one part in a normal string, one part in an interpolated string?